org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/XMLExcludedSymbolFile.java

```diff
@@ -0,0 +1,108 @@
+/*******************************************************************************
+* Copyright (c) 2019 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.lsp4xml.settings;
+
+import java.net.URI;
+import java.nio.file.PathMatcher;
+import java.util.Objects;
+
+import org.eclipse.lsp4xml.commons.TextDocument;
+import org.eclipse.lsp4xml.utils.URIUtils;
+
+/**
+ * XMLExcludedSymbolFiles
+ */
+public class XMLExcludedSymbolFile {
+
+	private transient PathMatcher pathMatcher;
+
+	private String pattern;
+
+	public XMLExcludedSymbolFile(String pattern) {
+		setPattern(pattern);
+	}
+
+	
+	public PathMatcher getPathMatcher() {
+		return pathMatcher;
+	}
+
+	public void setPathMatcher(PathMatcher pathMatcher) {
+		this.pathMatcher = pathMatcher;
+	}
+
+	public String getPattern() {
+		return pattern;
+	}
+
+	public void setPattern(String pattern) {
+		this.pattern = pattern;
+	}
+
+	public boolean matches(String uri) {
+		return URIUtils.matches(pattern, uri, pathMatcher);
+	}
+
+	public boolean matches(URI uri) {
+		return URIUtils.matches(pattern, uri, pathMatcher);
+	}
+
+	public static boolean matches(XMLExcludedSymbolFile[] excluded, TextDocument document) {
```

Comment:
Move this static method to XMLSymbolSettings (no static) isExcluded(String uri) method